src/v2/Apps/Settings/Routes/EditProfile/Components/SettingsEditProfileArtistsYouCollect/useAddArtistYouCollect.ts

```diff
@@ -0,0 +1,45 @@
+import { commitMutation, graphql } from "relay-runtime"
+import { useSystemContext } from "v2/System"
+import { useAddArtistYouCollectMutation } from "v2/__generated__/useAddArtistYouCollectMutation.graphql"
+
+export const useAddArtistYouCollect = () => {
+  const { relayEnvironment } = useSystemContext()
+
+  const submitAddArtistYouCollect = (artistID: string) => {
+    return new Promise((resolve, reject) => {
+      commitMutation<useAddArtistYouCollectMutation>(relayEnvironment!, {
+        onError: reject,
+        onCompleted: (res, errors) => {
+          if (errors !== null) {
+            reject(errors)
+            return
+          }
+
+          resolve(res)
+        },
+        mutation: graphql`
+          mutation useAddArtistYouCollectMutation(
+            $input: CreateUserInterestMutationInput!
+          ) {
+            createUserInterest(input: $input) {
+              clientMutationId
+              me {
+                ...SettingsEditProfileArtistsYouCollect_me
+                ...SettingsEditProfileYourGalleryIntro_me
```

Comment:
You'll notice the intro updates as well — love that we can just include fragments in the mutation to update non-local parts of the UI.